File: sdks/web/src/event-model.ts
export type EventType = 'track' | 'identify' | 'screen';

export interface DeviceContext {
  app_version: string;
  os_name: string;
  os_version: string;
  device_model: string;
  device_id: string;
  network_type: string;
  locale: string;
  timezone: string;
  sdk_version: string;
  screen_width: number;
  screen_height: number;
}

export interface TrueSightEvent {
  event_id: string;
  event_name: string;
  event_type: EventType;
  user_id: string | null;
  anonymous_id: string;
  mobile_number: string | null;
  email: string | null;
  client_timestamp: string;
  properties: Record<string, unknown>;
  context: DeviceContext;
}

export interface BatchPayload {
  batch: TrueSightEvent[];
  sent_at: string;
}

File: sdks/kmm/shared/src/commonMain/kotlin/com/truesight/sdk/EventModel.kt
package com.truesight.sdk

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Serializable
data class EventModel(
    @SerialName("event_id") val eventId: String,
    @SerialName("event_name") val eventName: String,
    @SerialName("event_type") val eventType: EventType,
    @SerialName("user_id") val userId: String? = null,
    @SerialName("anonymous_id") val anonymousId: String,
    @SerialName("mobile_number") val mobileNumber: String? = null,
    @SerialName("email") val email: String? = null,
    @SerialName("client_timestamp") val clientTimestamp: String,
    val properties: Map<String, JsonElement> = emptyMap(),
    val context: DeviceContextModel
)

@Serializable
enum class EventType {
    @SerialName("track") TRACK,
    @SerialName("identify") IDENTIFY,
    @SerialName("screen") SCREEN
}
